### PR TITLE
✨Add a close button to amp-ad stick type

### PR DESCRIPTION
**Pull request by powerivq**
The button will render when the ad is rendered the first time.

It also adds tests to cover https://github.com/ampproject/amphtml/pull/30306

### CODE REVIEW

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js [hunk 1]
```diff
@@ -47,7 +48,9 @@ import {
   getConsentPolicyState,
 } from '../../../src/consent';
 import {getIframe, preloadBootstrap} from '../../../src/3p-frame';
+import {listen} from '../../../src/event-helper';
 import {moveLayoutRect} from '../../../src/layout-rect';
+
```
Comment (calebcordry) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js:53]: nit: delete

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js [hunk 2]
```diff
@@ -455,6 +472,9 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override  */
   unlayoutCallback() {
+    this.unlisteners_.forEach((unlisten) => unlisten());
+    this.unlisteners_.length = 0;
```
Comment (calebcordry) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js:476]: should you set it back to empty array instead of overwriting length? can you even overwrite length?

**Review by calebcordry — APPROVED** [review]
LGTM other than length question.

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js (hunk 2 above)
Comment (powerivq) [reply extensions/amp-ad/0.1/amp-ad-3p-impl.js:476]: resolved offline. it's an officially supported way.

**Pull request merged**